### ISSUE TITLE
Rule proposal: `template-indent`

### ISSUE THREAD

**Issue by mmkal** [sindresorhus/eslint-plugin-unicorn#1469]
Tagged templates often look ugly/jarring because their indentation often doesn't match the code they're found in. In most cases, whitespace is insignificant, or a library like [strip-indent](https://www.npmjs.com/package/strip-indent) is used to remove the margin. See [proposal-string-dedent](https://github.com/tc39/proposal-string-dedent) (stage 1 at time of writing) for a proposal for fixing this in javascript.

## Fail

```js
function foo() {
  const a = `
abc
  def
`;

  const b = sql`
select *
from students
where first_name = ${x}
and last_name = ${y}
  `;

  // if you "fix" indentation manually, then copy code somewhere else, it can look stupid
  const c = gql`
                  query user(id: 5) {
                    firstName
                    lastName
                  }
                `;
}
```

## Pass

```js
function foo() {
  const a = `
    abc
      def
  `;

  const b = sql`
    select *
    from students
    where (first_name = ${x} or first_name = ${y})
    and last_name = ${z}
  `;

  const c = gql`
    query user(id: 5) {
      firstName
      lastName
    }
  `;
}
```

## Configuration

Of course in some templates are whitespace-sensitive, so by default this rule should leave them alone. I'd suggest a configuration option which tells the rule which template tags are safe to indent:

```js
{
  'unicorn/template-indent': ['warn', {
    tags: ['dedent'],
  }]
}
```

End-users could customize to add whichever tags they use in their codebase, including `null` if they want raw templates to be affected:

```js
{
  'unicorn/template-indent': ['warn', {
    tags: ['dedent', 'gql', 'sql', null],
  }]
}
```

## Details

Some bikeshedding might be needed, but my initial thoughts on how it should work:

- ignore templates that start and end on the same line
- for multiline templates:
  - look at the indentation of the line that the template begins on
  - ensure the template begins with a newline
  - have the second line begin one level further indented than the line of code the template starts on
  - convert spaces to tabs or vice versa based on the containing code
  - otherwise, preserve relative indentation of the template
- include in the documentation that it's ok to suppress this rule

**Comment by fisker** [comment]
There is already a proposal https://github.com/sindresorhus/eslint-plugin-unicorn/issues/580

**Comment by mmkal** [comment]
Thank you! My brain is clearly leakt, I gave that issue a 👍 at some point! 🤦
